src/main/java/com/mineboundteam/minebound/magic/PassiveSpellItem.java

```diff
@@ -26,7 +46,39 @@ protected static <T extends PassiveSpellItem> List<T> getEquippedSpellsOfType(Cl
                         spells.add((T) item.getItem());
                 }
             });
-
         return spells;
     }
+
+    @SuppressWarnings("unchecked")
+    protected static <T extends PassiveSpellItem> ItemStack getHighestEquippedSpellOfType(Class<T> type, List<ItemStack> spells) {
+        ItemStack highestSpell = null;
+        for (ItemStack spell : spells) {
+            if (highestSpell == null) {
+                highestSpell = spell;
+            } else if (((T) spell.getItem()).level.getValue() > ((T) highestSpell.getItem()).level.getValue()) {
+                highestSpell = spell;
+            }
+        }
+        return highestSpell;
+    }
+
+    protected static <T extends PassiveSpellItem> T getHighestEquippedSpellItem(List<T> spells) {
```

Comment:
The method name is misleading here; the list of spells doesn't need to have been equipped to a player necessarily

Reply:
Updated.